tools/ctl/ide/ui.go

```diff
@@ -305,9 +308,19 @@ func (e *explorer) showLogs(ctx context.Context, example rundex.Rebuild) {
 }
 
 func (e *explorer) editAndRun(ctx context.Context, example rundex.Rebuild) error {
-	localAssets, err := localfiles.AssetStore(example.RunID)
-	if err != nil {
-		return errors.Wrap(err, "failed to create local asset store")
+	var err error
+	var localAssets *rebuild.FilesystemAssetStore
+	if e.buildDefDir != "" {
+		fs, err := osfs.New("/").Chroot(e.buildDefDir)
```

Comment:
not a fan of osfs this deep in an callstack. if we can, injecting / constructing this further up would be preferable. makes testing easier and parameterizes fs access as much as possible.

Reply:
Instead of plumbing the buildDefDir all the way through, I changed to plumbing an buildDef LocatableAssetStore. This is one level higher than parameterizing fs, and I think it's closer to the level of abstraction the TUI actually cares about.